src/enums/index.ts

```diff
@@ -1,138 +1,153 @@
-import { makeClass } from "./../utils/make-class";
-const assert = require("assert");
-const _ = require("lodash");
-const { parseBytes, serializeUIntN } = require("./../utils/bytes-utils");
-const enums = require("./definitions.json");
-
-function transformWith(func, obj) {
-  return _.transform(obj, func);
+import { serializeUIntN } from "./../utils/bytes-utils";
+import * as enums from "./definitions.json";
+
+const TYPE_WIDTH = 2;
+const LEDGER_ENTRY_WIDTH = 2;
+const TXN_TYPE_WIDTH = 2;
+const TXN_RESULT_WIDTH = 1;
+
+class EnumType {
+  bytes: Uint8Array;
+
+  constructor(
+    public name: string,
+    public ordinal: number,
+    public ordinalWidth: number
+  ) {
+    this.bytes = serializeUIntN(ordinal, ordinalWidth);
+  }
+
+  toJSON(): string {
+    return this.name;
+  }
+
+  toBytesSink(sink): void {
+    sink.put(this.bytes);
+  }
+}
+
+class EnumTypes {
+  constructor(types: { [key: string]: number }, public ordinalWidth: number) {
+    Object.entries(types).forEach(([k, v]) => {
+      this[k] = new EnumType(k, v, ordinalWidth);
+      this[v.toString()] = this[k];
+    });
+  }
+
+  from(value: EnumType | string): EnumType {
```

Comment:
As a non-JS programmer, I find polymorphic APIs to be somewhat hard to reason about. 

You already have `fromParser`, why not `fromEnum` and `fromString`?

Reply:
Thats a good idea. Will also have to change up some of the other fromParser definitions as code is refactored